src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ModelDirective.cs

```diff
@@ -66,6 +73,19 @@ public override void ExecuteCore(RazorCodeDocument codeDocument, DocumentIRNode
                 var visitor = new Visitor();
                 var modelType = GetModelType(irDocument, visitor);
 
+                if (_designTime)
+                {
+                    // This is needed to supress compilation errors about unknown TModel type.
+                    var @namespace = visitor.Class.Parent;
```

Comment:
Lets not assume that the classes parent is the namespace. You can just raise the namespace in the visitor.